Language Server crash when variable kind is not specified

### Problem

The Cadence language server crashes while processing a contract containing an embedded optional resource with the VS Code extension. 

I am using:
* Flow CLI, installed from source at https://github.com/onflow/flow-cli/commit/2406a02b25691c495e1e321b15878ac882952513
  * This appears to be using Cadence `v0.9.1`
* VS Code extension version `v0.5.0`, installed with `flow-cli`

### Steps to Reproduce

The Language Server crashes with the following code when typing either the `o`, `t`, or `e` in `voter` in the line `destroy self.voter`. If I restart the Language Server with `self.voter` fully or partially typed out, it doesn't crash. 

To reproduce:
* Copy `EpochClusterQCs` contract to a VS Code document
* Delete `voter` from the line marked `RELEVANT LINE`
* Re-type `voter`

```
pub contract EpochClusterQCs {

    pub resource QCVoter {}

    pub resource QCVoterReceiver {
        pub voter: @EpochClusterQCs.QCVoter?

        init() {
            self.voter <- nil
        }

        destroy() {
            if self.voter != nil {
                destroy self.voter      // RELEVANT LINE
            }
        }
    }
}
```


Trace:
```
panic: unreachable
goroutine 8 [running]:
runtime/debug.Stack(0xc00007d090, 0xc000374d00, 0x18)
	/usr/local/Cellar/go/1.14.4/libexec/src/runtime/debug/stack.go:24 +0x9d
github.com/onflow/cadence/runtime/errors.NewUnreachableError(...)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence@v0.9.1/runtime/errors/errors.go:44
github.com/onflow/cadence/runtime/ast.VariableKind.Name(...)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence@v0.9.1/runtime/ast/variablekind.go:54
github.com/onflow/cadence/languageserver/server.(*Server).ResolveCompletionItem(0xc000438580, 0x50ad280, 0xc0004020d8, 0xc0002b3930, 0xc0002b3930, 0x0, 0x0)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence/languageserver@v0.9.4/server/server.go:994 +0x853
github.com/onflow/cadence/languageserver/protocol.(*Server).handleCompletionItemResolve(0xc000424e40, 0xc000129880, 0xc0001a8010, 0xc000442388, 0x0, 0xc000442340)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence/languageserver@v0.9.4/protocol/methods.go:103 +0xcc
github.com/onflow/cadence/languageserver/jsonrpc2.(*handler).Handle(0xc000402100, 0x50ad080, 0xc000134008, 0xc00043a240, 0xc0000aaf00)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence/languageserver@v0.9.4/jsonrpc2/server.go:53 +0x2dd
github.com/sourcegraph/jsonrpc2.(*Conn).readMessages(0xc00043a240, 0x50ad080, 0xc000134008)
	/Users/dapperlabs/dev/pkg/mod/github.com/sourcegraph/jsonrpc2@v0.0.0-20191222043438-96c4efab7ee2/jsonrpc2.go:522 +0x567
created by github.com/sourcegraph/jsonrpc2.NewConn
	/Users/dapperlabs/dev/pkg/mod/github.com/sourcegraph/jsonrpc2@v0.0.0-20191222043438-96c4efab7ee2/jsonrpc2.go:334 +0x1cc


goroutine 8 [running]:
github.com/onflow/cadence/runtime/ast.VariableKind.Name(...)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence@v0.9.1/runtime/ast/variablekind.go:54
github.com/onflow/cadence/languageserver/server.(*Server).ResolveCompletionItem(0xc000438580, 0x50ad280, 0xc0004020d8, 0xc0002b3930, 0xc0002b3930, 0x0, 0x0)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence/languageserver@v0.9.4/server/server.go:994 +0x9b1
github.com/onflow/cadence/languageserver/protocol.(*Server).handleCompletionItemResolve(0xc000424e40, 0xc000129880, 0xc0001a8010, 0xc000442388, 0x0, 0xc000442340)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence/languageserver@v0.9.4/protocol/methods.go:103 +0xcc
github.com/onflow/cadence/languageserver/jsonrpc2.(*handler).Handle(0xc000402100, 0x50ad080, 0xc000134008, 0xc00043a240, 0xc0000aaf00)
	/Users/dapperlabs/dev/pkg/mod/github.com/onflow/cadence/languageserver@v0.9.4/jsonrpc2/server.go:53 +0x2dd
github.com/sourcegraph/jsonrpc2.(*Conn).readMessages(0xc00043a240, 0x50ad080, 0xc000134008)
	/Users/dapperlabs/dev/pkg/mod/github.com/sourcegraph/jsonrpc2@v0.0.0-20191222043438-96c4efab7ee2/jsonrpc2.go:522 +0x567
created by github.com/sourcegraph/jsonrpc2.NewConn
	/Users/dapperlabs/dev/pkg/mod/github.com/sourcegraph/jsonrpc2@v0.0.0-20191222043438-96c4efab7ee2/jsonrpc2.go:334 +0x1c
```

### Acceptance Criteria

<if any>

### Context

<what are you currently working on that this is blocking?>

--- a/languageserver/server/server.go
+++ b/languageserver/server/server.go
@@ -1035,12 +1035,19 @@ func (s *Server) ResolveCompletionItem(
 		typeString := member.TypeAnnotation.Type.QualifiedString()
 
 		result.Detail = fmt.Sprintf(
-			"(%s) %s.%s: %s",
-			member.VariableKind.Name(),
+			"%s.%s: %s",
 			member.ContainerType.String(),
 			member.Identifier,
 			typeString,
 		)
+
+		// add the variable kind, if any, as a prefix
+		if member.VariableKind != ast.VariableKindNotSpecified {
+			result.Detail = fmt.Sprintf("(%s) %s",
+				member.VariableKind.Name(),
+				result.Detail,
+			)
+		}
 
 	case common.DeclarationKindFunction:
 		typeString := member.TypeAnnotation.Type.QualifiedString()

--- a/runtime/ast/variablekind.go
+++ b/runtime/ast/variablekind.go
@@ -45,6 +45,8 @@ var VariableKinds = []VariableKind{
 
 func (k VariableKind) Name() string {
 	switch k {
+	case VariableKindNotSpecified:
+		return ""
 	case VariableKindVariable:
 		return "variable"
 	case VariableKindConstant:
@@ -56,6 +58,8 @@ func (k VariableKind) Name() string {
 
 func (k VariableKind) Keyword() string {
 	switch k {
+	case VariableKindNotSpecified:
+		return ""
 	case VariableKindVariable:
 		return "var"
 	case VariableKindConstant:

